scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,824 @@
+            throws InvalidChannelException {
+        return getChannel(channel).containsKey(key);
+    }
+
+    @Override
+    public int size(String originator, TaskId taskid, String channel) throws InvalidChannelException {
+        return getChannel(channel).size();
+    }
+
+    @Override
+    public boolean isEmpty(String originator, TaskId taskid, String channel) throws InvalidChannelException {
+        return getChannel(channel).isEmpty();
+    }
+
+    @Override
+    public boolean containsValue(String originator, TaskId taskid, String channel, Serializable value)
+            throws InvalidChannelException {
+        return getChannel(channel).containsValue(value);
+    }
+
+    @Override
+    public <T extends Serializable> T get(String originator, TaskId taskid, String channel, String key)
+            throws InvalidChannelException {
+        return ((Channel<T>) getChannel(channel)).get(key);
+    }
+
+    @Override
+    public <T extends Serializable> T put(String originator, TaskId taskid, String channel, String key, T value)
+            throws InvalidChannelException, IOException {
+        T answer = ((Channel<T>) getChannel(channel)).put(key, value);
+        commitIfNeeded(channel);
+        logWithContext(taskid,
+                       channel,
+                       "Put " + value + " on key '" + key + "', previous value was " + answer + "",
+                       Level.DEBUG);
+        return answer;
+    }
+
+    @Override
+    public <T extends Serializable> T remove(String originator, TaskId taskid, String channel, String key)
+            throws InvalidChannelException, IOException {
+        T answer = ((Channel<T>) getChannel(channel)).remove(key);
+        commitIfNeeded(channel);
+        logWithContext(taskid, channel, "Removed key '" + key + "', previous value was " + answer, Level.DEBUG);
+        return answer;
+    }
+
+    @Override
+    public <T extends Serializable> void putAll(String originator, TaskId taskid, String channel,
+            Map<? extends String, ? extends T> m) throws InvalidChannelException, IOException {
+        ((Channel<T>) getChannel(channel)).putAll(m);
+        logWithContext(taskid, channel, "PutAll called, with new entries: " + m, Level.DEBUG);
+        commitIfNeeded(channel);
+    }
+
+    @Override
+    public void clear(String originator, TaskId taskid, String channel) throws InvalidChannelException, IOException {
+        getChannel(channel).clear();
+        logWithContext(taskid, channel, "cleared channel", Level.DEBUG);
+        commitIfNeeded(channel);
+    }
+
+    @Override
+    public Set<String> keySet(String originator, TaskId taskid, String channel) throws InvalidChannelException {
+        return getChannel(channel).keySet();
+    }
+
+    @Override
+    public <T extends Serializable> Collection<T> values(String originator, TaskId taskid, String channel)
+            throws InvalidChannelException {
+        return ((Channel<T>) getChannel(channel)).values();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast to "Channel". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1905)